cmd/xl-zones.go

```diff
@@ -1533,9 +1534,58 @@ func (z *xlZones) GetMetrics(ctx context.Context) (*Metrics, error) {
 	return &Metrics{}, NotImplemented{}
 }
 
+func (z *xlZones) getZoneAndSet(id string) (int, int, error) {
+	for zoneIdx := range z.zones {
+		format := z.zones[zoneIdx].format
+		for setIdx, set := range format.XL.Sets {
+			for _, diskID := range set {
+				if diskID == id {
+					return zoneIdx, setIdx, nil
+				}
+			}
+		}
+	}
+	return 0, 0, errDiskNotFound
+}
+
 // IsReady - Returns true if first zone returns true
 func (z *xlZones) IsReady(ctx context.Context) bool {
-	return z.zones[0].IsReady(ctx)
```

Comment:
Perhaps you should purge this implementation and return `true` always? @krishnasrinivas to avoid mis-use

Reply:
now it returns true for liveness, this is ready. Did you mean it for liveness?

Reply:
No I meant xl-sets implementation can be removed..